app/authentication/auth.py

```diff
@@ -102,6 +107,10 @@ def requires_auth():
         except TokenDecodeError:
             continue
         except TokenExpiredError:
+            decoded_token = decode_token(auth_token)
```

Comment:
`decode_token` can raise `DecodeError` if the json.loads(...) from the decoded text throws, it can also throw a `TypeError`. Are we OK in handling these exceptions in the parent caller chain?

Reply:
Caught a potential `DecodeError` in the last commit. I don't think running `decode_token` every time in the `try` statement is needed, would like to run this only if this rare error happens.